server/filestore_test.go

```diff
@@ -37,483 +37,499 @@ import (
 	"github.com/klauspost/compress/s2"
 )
 
-func TestFileStoreBasics(t *testing.T) {
-	storeDir := t.TempDir()
-
-	fs, err := newFileStore(FileStoreConfig{StoreDir: storeDir}, StreamConfig{Name: "zzz", Storage: FileStorage})
-	if err != nil {
-		t.Fatalf("Unexpected error: %v", err)
+func testFileStoreAllPermutations(t *testing.T, fn func(t *testing.T, fcfg FileStoreConfig)) {
```

Comment:
maybe `testFileStoreAllCiphers` ?

Reply:
Main reason why I went with this name is because eventually it will be expanded to compression too, i.e.
```
	for _, fcfg := range []FileStoreConfig{
		{Cipher: ChaCha, Compression: NoCompression},
		{Cipher: ChaCha, Compression: GZIPCompression},
		{Cipher: ChaCha, Compression: S2Compression},
		{Cipher: AES, Compression: NoCompression},
		{Cipher: AES, Compression: GZIPCompression},
		{Cipher: AES, Compression: S2Compression},
	} {
```

Reply:
ah, then `Permutation` fits better. Ok.